src/Core/src/Platform/Android/ViewExtensions.cs

```diff
@@ -93,7 +93,19 @@ public static void Unfocus(this AView platformView, IView view)
 
 		public static void UpdateVisibility(this AView platformView, IView view)
 		{
-			platformView.Visibility = view.Visibility.ToPlatformVisibility();
+			if (platformView is WrapperView wrapperView)
+				wrapperView.UpdateVisibility(view);
+			else
+				platformView.Visibility = view.Visibility.ToPlatformVisibility();
+		}
+
+		internal static void UpdateVisibility(this WrapperView wrapperView, IView view)
+		{
+			wrapperView.Visibility = view.Visibility.ToPlatformVisibility();
+
+			for (int i = 0; i < wrapperView.ChildCount; i++)
+				if (wrapperView.GetChildAt(i) is AView child)
+					child.Visibility = wrapperView.Visibility;
```

Comment:
Good question. Mnnn, we use WrapperView for exceptional cases where we need to wrap a View to get effects like Shadows etc. Normally WrapperView only contain a child element along with adorners etc (drawables, layers etc). The WrapperView is not used with other handlers, nor are children added from the abstraction. At the framework level we add/remove the WrapperView dynamically. When we want to hide a Button, if it needs certain effects (eg Shadow), we will have WrapperView -> Button. If we want to change the visibility, etc. everything applied to the WrapperView we want to propagate to the Button.

Reply:
Are we propagating it to the button so that anything else inspecting that platform button can infer the right visibility state?

In other words, do we need to propagate it to the button?  The wrapperview being invisible will cause the button to also be invisible, correct?

Reply:
Right, the invisible WrapperView will change the Button visibility to invisible too.